plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java

```diff
@@ -91,14 +99,55 @@ public synchronized void checkLimitAndAddRunningTask(MLTask mlTask, Integer limi
             throw new MLLimitExceededException(error);
         }
         if (contains(mlTask.getTaskId())) {
-            getMLTask(mlTask.getTaskId()).setState(MLTaskState.RUNNING);
+            getMLTask(mlTask.getTaskId()).setState(RUNNING);
         } else {
-            mlTask.setState(MLTaskState.RUNNING);
+            mlTask.setState(RUNNING);
             add(mlTask);
         }
         runningTaskCount.incrementAndGet();
     }
 
+    public synchronized void checkMaxBatchJobTask(MLTaskType mlTaskType, Integer maxTaskLimit, ActionListener<Boolean> listener) {
+        try {
+            BoolQueryBuilder boolQuery = QueryBuilders
+                .boolQuery()
+                .must(QueryBuilders.termQuery(TASK_TYPE_FIELD, mlTaskType.name()))
+                .must(
+                    QueryBuilders
+                        .boolQuery()
+                        .should(QueryBuilders.termQuery(STATE_FIELD, CREATED))
+                        .should(QueryBuilders.termQuery(STATE_FIELD, RUNNING))
+                );
+
+            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(boolQuery);
```

Comment:
Since we only need total hits, suggest add size as 0 to avoid returning documents detail content.
```
searchSourceBuilder.size(0);
```